openedx/core/djangoapps/courseware_api/tests/test_views.py

```diff
@@ -430,17 +432,16 @@ def test_can_access_proctored_exams_masquerading(self, masquerade_group_id, resu
         assert 'can_access_proctored_exams' in courseware_data
         assert courseware_data['can_access_proctored_exams'] == result
 
-    @override_waffle_flag(COURSEWARE_LEARNING_ASSISTANT, active=False)
-    def test_learning_assistant_enabled_disabled_waffle_flag(self):
-        response = self.client.get(self.url)
-        learning_assistant_enabled = response.json()['learning_assistant_enabled']
-        self.assertFalse(learning_assistant_enabled)
+    @ddt.idata(itertools.product((True, False), (True, False)))
+    @ddt.unpack
+    def test_learning_assistant_enabled(self, setting_enabled, flag_enabled):
+        with override_settings(LEARNING_ASSISTANT_AVAILABLE=setting_enabled):
```

Comment:
Nit.

You don't have to nest the context managers, if you don't want to. You can use them on [one line](https://stackoverflow.com/a/3025119).